lib/osv/purl_helpers.py

```diff
@@ -27,6 +29,12 @@
 }
 
 
+def _url_encode(package_name):
+  """URL encode a PURL `namespace/name` or `name`."""
+  parts = package_name.split('/')
+  return '/'.join(quote(p) for p in parts)
```

Comment:
Are there possibly qualifiers or subpath in the package name purls? If so this will also % encode the ? and # separators.

Reply:
Good question! PURLs do support qualifiers (https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst#package-url-specification-v10x)

However, we don't generate them currently, so there's no need to account for them.